### PR TITLE
Iris l elevator

### PR DESCRIPTION
**Pull request by ChainLynx-Robotics**
It works

### CODE REVIEW

**Comment by ChainLynx-Robotics** [comment]
Elevator setpoints functional, manual override button implemented.

**Review by ThatQuadra — APPROVED** [review]
Looking snazzy

File: src/main/java/frc/robot/Commands/ElevatorCommand.java [hunk 1]
```diff
@@ -46,7 +47,24 @@ public void execute() {
 
         System.out.printf("setting setpoint via command, setpoint at: %f \n", setpoint);
         SmartDashboard.putNumber("Elevator Setpoint (rotations)", setpoint);
-        elevator.moveElevator(setpoint);
+
+        if (setpoint-elevator.getDrivingEncoder().getPosition() > 8) {
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorCommand.java:51]: What's with all the hard-coded numbers? Also, what's the purpose of this math?

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -84,6 +86,12 @@ public void teleopInit() {
         if (m_autonomousCommand != null) {
             m_autonomousCommand.cancel();
         }
+
+        CommandScheduler.getInstance().schedule(
+          new ElevatorCommand(m_robotContainer.getElevator(), m_robotContainer.getIntake(),14));
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Robot.java:91]: What's this hard-coded number?

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -189,6 +189,8 @@ public class ElevatorConstants {
       public static final double highElevatorCubeSetpoint = 13.5;
       public static final double doubleSubstationSetpoint = 14;
 
+      public static final double ELEVATOR_RAMP_DIST = 3;
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Constants.java:192]: What is this?

File: src/main/java/frc/robot/Commands/ElevatorCommand.java [hunk 4]
```diff
@@ -46,7 +47,24 @@ public void execute() {
...
+            elevator.moveElevator(setpoint-ElevatorConstants.ELEVATOR_RAMP_DIST);
+            double startTime = System.currentTimeMillis();
+
+            if (System.currentTimeMillis()-startTime > 100) {
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorCommand.java:55]: What is this?

File: src/main/java/frc/robot/Commands/ElevatorCommand.java [hunk 5]
```diff
@@ -46,7 +47,24 @@ public void execute() {
...
+                elevator.moveElevator(setpoint);
+            }
+        } else if (setpoint-elevator.getDrivingEncoder().getPosition() < -8) {
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorCommand.java:58]: Magic number -8 means what?

File: src/main/java/frc/robot/Commands/ElevatorCommand.java [hunk 6]
```diff
@@ -46,7 +47,24 @@ public void execute() {
...
+            elevator.moveElevator(setpoint+ElevatorConstants.ELEVATOR_RAMP_DIST);
+            double startTime2 = System.currentTimeMillis();
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorCommand.java:60]: Why startTime2?

File: src/main/java/frc/robot/Commands/ElevatorControlCommand.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    public ElevatorControlCommand(Joystick stick, ElevatorSubsystem elevator) {
+        this.elevator = elevator;
+        this.stick = stick;
+
+        addRequirements(elevator);
+    }
+
+    @Override
+    public void execute() {
+        elevator.simpleMovement(stick.getRawAxis(1));
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Commands/ElevatorControlCommand.java:20]: Replace magic number with constant

File: src/main/java/frc/robot/RobotContainer.java [hunk 8]
```diff
@@ -147,12 +139,11 @@ private void configureButtonBindings() {
         )
       );
 
-    new Trigger(() -> m_operatorController.getRawButton(m_operatorController.getPOV(0)))
-      .onTrue(new InstantCommand(m_arm::retract));
-
-    new Trigger(() -> m_operatorController.getRawButton(m_operatorController.getPOV(180)))
-      .onTrue(new InstantCommand(m_arm::expand));
+    POVButton upPOV = new POVButton(m_operatorController, 0);
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/RobotContainer.java:142]: Replace magic number with a constant

File: src/main/java/frc/robot/RobotContainer.java [hunk 9]
```diff
@@ -198,15 +188,8 @@ private void configureButtonBindings() {
...
-=======
-
-    new Trigger(() -> m_operatorController.getRawButton(Bindings.fullRetraction))
-      .onTrue(
-        new ElevatorCommand(
-          m_elevator,
-          m_intake,
-          Bindings.fullRetraction));
->>>>>>> 4c014fc4b12b5c00e5ad54f70e6c9e25a37f3c9b
+    new Trigger(() -> m_operatorController.getRawButton(5))
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/RobotContainer.java:191]: Magic number -> constant

File: src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java [hunk 10]
```diff
@@ -157,6 +157,7 @@ public void moveElevator(Double setpoint) {
     }
 
     public void simpleMovement(double input) {
+        System.out.println("simple movement running");
         if (input > 0.25) {
```
Comment (garrettwitowski) [inline src/main/java/frc/robot/Subsystems/ElevatorSubsystem.java:161]: Magic number -> constant

**Review by garrettwitowski — COMMENTED** [review]
Main question is around all the hard-coded values in ElevatorCommand.

**Pull request merged**